File: Leet Code/56_merge_intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        if not intervals:
            return intervals
        n = len(intervals)
        intervals.sort()
        prev = intervals[0]
        ans = []
        for i in range(1,n):
            if prev[1] >= intervals[i][0]:
                if prev[0] > intervals[i][0]:
                    prev[0] = intervals[i][0]
                
                if prev[1] < intervals[i][1]:
                    prev[1] = intervals[i][1]
            else:
                ans.append(prev)
                prev = intervals[i]
        ans.append(prev)
        
        return ans 
                

File: Leet Code/56_merge_intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        int n = intervals.size();
        if(n == 1 || n == 0) return intervals;
        sort(intervals.begin(), intervals.end());
        vector<vector<int>> ans;
        ans.push_back(intervals[0]);
        int j = 0;
        for(int i = 0; i < n; i++){
            if(ans[j][1] >= intervals[i][0]){
                if(ans[j][0] > intervals[i][0]){
                    ans[j][0] = intervals[i][0];
                }
                if(ans[j][1] < intervals[i][1]){
                    ans[j][1] = intervals[i][1];
                }
            }else{
                ans.push_back(intervals[i]);
                j++;
            }
        }
        return ans;
    }
};